### PR TITLE
Add a url field to messages and add .setCategory function for guild channels

### PR DESCRIPTION
**Pull request by oq-x**
This pull requests adds a url field for the Message structure which is a link for the message
and adds the .setCategory function for all guild channels

### CODE REVIEW

File: src/structures/message.ts [hunk 1]
```diff
@@ -115,6 +116,7 @@ export class Message extends SnowflakeBase {
     this.application = data.application ?? this.application
     this.messageReference = data.message_reference ?? this.messageReference
     this.flags = data.flags ?? this.flags
+    this.url = `https://discord.com/channels/${data.guild_id}/${data.channel_id}/${this.id}`
```
[the remark this answers is not in the document]
Comment (oq-x) [reply src/structures/message.ts]: fixed

File: src/structures/message.ts [hunk 2]
```diff
@@ -115,6 +116,7 @@ export class Message extends SnowflakeBase {
     this.application = data.application ?? this.application
     this.messageReference = data.message_reference ?? this.messageReference
     this.flags = data.flags ?? this.flags
+    this.url = `https://discord.com/channels/${data.guild_id || "@me"}/${data.channel_id}/${this.id}`
```
Comment (Helloyunho) [inline src/structures/message.ts:119]: i think you should use `this.guildID` and `this.channelID` for this because data could also have no channel_id

**Review by Helloyunho — CHANGES_REQUESTED** [review]

File: src/structures/message.ts (hunk 2 above)
Comment (oq-x) [reply src/structures/message.ts]: fixed

**Comment by cubedhuang** [comment]
I feel like this should be a getter rather than an eagerly set property, similar User#mention and User#tag

**Comment by oq-x** [comment]
I can't right now

**Comment by Helloyunho** [comment]
kay I'll do it

**Review by Helloyunho — APPROVED** [review]
lgtm

**Pull request merged**